doc/source/ray-air/examples/batch_tuning.ipynb

```diff
@@ -0,0 +1,1154 @@
+    "\n",
+    "**In the cell below, our \"Trainable\" function is called `train_model`**.  It takes as input a config dictionary argument, and returns a simple dict output. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "a03c0ff8-0c42-4703-91f4-57fc0df8f0ea",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 1. Define a custom train function\n",
+    "def train_model(config: dict):\n",
+    "\n",
+    "    model = config['model']\n",
+    "    the_location = config['location']\n",
+    "    \n",
+    "    # Load data.\n",
+    "    df_list = [read_data(f, the_location) for f in s3_files]   \n",
+    "    df_raw = pd.concat(df_list, ignore_index=True)\n",
+    "    df = transform_df(df_raw)\n",
+    "    \n",
+    "    # Train/test split.\n",
+    "    train_df, test_df = train_test_split(df, test_size=0.2, shuffle=True)\n",
+    "    train_X = train_df[[\"passenger_count\", \"trip_distance\"]]\n",
+    "    train_y = train_df.trip_duration\n",
+    "    test_X = test_df[[\"passenger_count\", \"trip_distance\"]]\n",
+    "    test_y = test_df.trip_duration\n",
+    "\n",
+    "    # Train model.\n",
+    "    model = model.fit(train_X, train_y)\n",
+    "    pred_y = model.predict(test_X)\n",
+    "    \n",
+    "    # Evaluate.\n",
+    "    error = sklearn.metrics.mean_absolute_error(test_y, pred_y)\n",
+    "    \n",
+    "    # Save the model as a Tune Checkpoint. This is done through \n",
+    "    # ray.air.session.report() API, which can take in a Checkpoint object.\n",
+    "    # https://docs.ray.io/en/latest/tune/tutorials/tune-checkpoints.html\n",
+    "    \n",
+    "    # Save the model checkpoint\n",
+    "    checkpoint = SklearnCheckpoint.from_estimator(model, path=\".\")\n",
+    "\n",
+    "    # Save the checkpoint using ray.air.session()\n",
+    "    session.report(\n",
+    "            {\"error\": error}, \n",
+    "            checkpoint=checkpoint)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "72369cbf-e346-441c-b7e3-9280ccabc7f6",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Define your Ray Tune Search Space <a class=\"anchor\" id=\"define_search_space\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e310b13e-ba83-4c47-84e6-d1e916483b2b",
+   "metadata": {},
+   "source": [
+    "**Next, define a search space of trials to run.** Below, we define a simple grid search of 2 Scikit-learn models and NYC taxi drop-off locations. \n",
+    "\n",
+    "Ray Tune will generate permutations of the grid search parameters and pass these in the config dictionary passed to each Trainable function call.\n",
+    "\n",
```

Comment:
we need to call out that we are using grid search to make sure that we generate every possible partition-model combination.